client/src/test/java/org/spine3/base/MismatchShould.java

```diff
@@ -61,28 +67,103 @@ public void return_default_found_value_if_it_was_passed_as_null() {
+    public void return_mismatch_object_with_string_values() {
+        final ValueMismatch mismatch = Mismatch.of(EXPECTED, ACTUAL, REQUESTED, VERSION);
         final StringValue expected = fromAny(mismatch.getExpected());
-        final StringValue found = fromAny(mismatch.getActual());
+        final StringValue actual = fromAny(mismatch.getActual());
         final StringValue requested = fromAny(mismatch.getRequested());
 
         assertEquals(EXPECTED, expected.getValue());
-        assertEquals(FOUND, found.getValue());
+        assertEquals(ACTUAL, actual.getValue());
         assertEquals(REQUESTED, requested.getValue());
     }
 
     @Test
-    public void return_default_expected_value_if_it_was_passed_as_null_message_overload() {
-        final ValueMismatch mismatch = Mismatch.of(null, newStringValue(FOUND), newStringValue(REQUESTED), VERSION);
+    public void return_mismatch_object_with_int32_values() {
+        final int value0 = 0;
+        final int value1 = 1;
+        final int value2 = 2;
+        final ValueMismatch mismatch = Mismatch.of(value0, value1, value2, VERSION);
+        final Int32Value expected = fromAny(mismatch.getExpected());
+        final Int32Value actual = fromAny(mismatch.getActual());
+        final Int32Value requested = fromAny(mismatch.getRequested());
+
+        assertEquals(value0, expected.getValue());
+        assertEquals(value1, actual.getValue());
+        assertEquals(value2, requested.getValue());
+    }
+
+    @Test
+    public void return_mismatch_object_with_int64_values() {
+        final long value0 = 0L;
+        final long value1 = 1L;
+        final long value2 = 2L;
+        final ValueMismatch mismatch = Mismatch.of(value0, value1, value2,  VERSION);
+        final Int64Value expected = fromAny(mismatch.getExpected());
+        final Int64Value actual = fromAny(mismatch.getActual());
+        final Int64Value requested = fromAny(mismatch.getRequested());
+
+        assertEquals(value0, expected.getValue());
+        assertEquals(value1, actual.getValue());
+        assertEquals(value2, requested.getValue());
+    }
+
+    @Test
+    public void return_mismatch_object_with_float_values() {
+        final float value0 = 0.0F;
+        final float value1 = 1.0F;
+        final float value2 = 2.0F;
+        final ValueMismatch mismatch = Mismatch.of(value0, value1, value2, VERSION);
+        final FloatValue expected = fromAny(mismatch.getExpected());
+        final FloatValue actual = fromAny(mismatch.getActual());
+        final FloatValue requested = fromAny(mismatch.getRequested());
+
+        assertEquals(value0, expected.getValue(), DELTA);
+        assertEquals(value1, actual.getValue(), DELTA);
+        assertEquals(value2, requested.getValue(), DELTA);
+    }
+
+    @Test
+    public void return_mismatch_object_with_double_values() {
+        final double value0 = 0.1;
+        final double value1 = 0.1;
```

Comment:
It's better to rename them to `expected`, `actual`, `requested`. And you can get rid of the existing variables by inlining them.